pkg/controller/telemetry/bpf_prog_metric.go

```diff
@@ -117,49 +116,45 @@ func (m *OperationMetricController) Run(ctx context.Context, KmeshPerfInfo *ebpf
 			operationData.endTime = binary.LittleEndian.Uint64(rec.RawSample[8:16])
 			operationData.pidTgid = binary.LittleEndian.Uint64(rec.RawSample[16:24])
 			operationData.operationType = binary.LittleEndian.Uint32(rec.RawSample[24:28])
-			metricLabels := m.buildOperationMetric(&operationData)
-			operationInfo := operationUsageInfo{
+			operationInfo := operationDuration{
 				durations:     []uint64{operationData.endTime - operationData.startTime},
 				operationType: operationData.operationType,
 			}
 			m.mutex.Lock()
-			m.updateOperationMetricCache(operationInfo, metricLabels)
+			m.updateOperationMetricCache(operationInfo, buildOperationMetricLabel(&operationData))
 			m.mutex.Unlock()
 		}
 	}
 }
 
-// buildOperationMetric builds the operation metrics labels using actual pod info.
-func (m *OperationMetricController) buildOperationMetric(data *operationUsageMetric) operationMetricLabels {
-	labels := operationMetricLabels{}
-	nodeName := os.Getenv("NODE_NAME")
-	if nodeName == "" {
-		nodeName = "unknown"
+// buildOperationMetricLabel builds the operation metrics labels using actual pod info.
+func buildOperationMetricLabel(data *operationTimeMetric) operationMetricLabels {
+	return operationMetricLabels{
+		nodeName:      os.Getenv("NODE_NAME"),
+		operationType: operationTypeMap[data.operationType],
 	}
-	labels.nodeName = nodeName
-	labels.operationType = operationTypeMap[data.operationType]
-	return labels
 }
 
-func (m *OperationMetricController) updateOperationMetricCache(data operationUsageInfo, labels operationMetricLabels) {
+func (m *BpfProgMetric) updateOperationMetricCache(data operationDuration, labels operationMetricLabels) {
+	m.mutex.Lock()
```

Comment:
Lock twice?

Reply:
Oh, i meant to move lock into hte function